packages/core/docs/rfc/service-request.md

```diff
@@ -0,0 +1,228 @@
+# Service Request
+   
+## Overview and Motivation
+The Service handles incoming messages via several action Transports. On each incoming message Transport Router Adapter 
+builds an object with `ServiceRequest` type and passes it to the Dispatcher. There is no common constructor function now. 
+This object becomes available in the Action Handler as an argument. Its structure is abstract and functionality is 
+basically transport-agnostic.
+
+
+Most transport protocols natively have headers in messages. While currently every Transport Router Adapter is able to 
+parse incoming messages including its optional headers, in order to fully support messaging protocols the Service should 
+provide a way to set, modify and remove response message headers. That's why we need to implement **common reply headers 
+API**.
```

Comment:
BTW I would prefer `responseHeaders` name instead of `replyHeaders` that is symmetric to `Request`.